exp/controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -54,13 +55,20 @@ func newAzureManagedMachinePoolReconciler(scope *scope.ManagedControlPlaneScope)
 // Reconcile reconciles all the services in pre determined order
 func (r *azureManagedMachinePoolReconciler) Reconcile(ctx context.Context, scope *scope.ManagedControlPlaneScope) error {
 	scope.Logger.Info("reconciling machine pool")
+
+	var normalizedVersion *string
+	if scope.MachinePool.Spec.Template.Spec.Version != nil {
```

Comment:
can this go in a webhook?

Reply:
Are you suggesting a default version?

Reply:
yes that way the normalized version appears in the spec and it's not too hidden from the user, same as https://github.com/CecileRobertMichon/cluster-api/blob/master/api/v1alpha3/machine_webhook.go#L60

Reply:
Webhooks are not currently setup for `AzureManged*`. Can we open an issue to track this and pull in these changes?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/615
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/612

I can add this to `AzureManagedMachinePool` in this PR.

Reply:
ah, actually in machine webhook it's defaulted inside the same object, this is cross object -- it's taking it from the upstream machine pool. so maybe we should fix it there, if we didn't already?

Reply:
Great point.

Reply:
That was part of the changes in https://github.com/kubernetes-sigs/cluster-api/pull/3147, the Version isn't part of MachinePool spec, it's part of Machine spec which is embedded in the MachinePool spec as part of the template https://github.com/kubernetes-sigs/cluster-api/blob/master/exp/api/v1alpha3/machinepool_types.go#L44

Reply:
So actually do we even need this normalization here? Since the Machine webhook already takes care of it

Reply:
For `AzureManagedControlPlane` Version _is_ part of the object https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/818/files#diff-69a80dafa12ba6b9eeaf233d609e8f5eR29 so we could do the same normalization in a webhook, for `AzureMachinePool` I believe it's already taken care of by the `Machine` webhooks

Reply:
Ok so after some coffee and thinking through it with @devigned I think we need to: 
1) keep the AzureMachinePool normalization until we can move it into the capi MachinePool webhook
2) Add a webhook normalization for AzureManagedControlPlane, either here or in a separate PR 

Let's track whatever we don't get done here in issues so they don't get lost.

I take back what I said above, the Machine webhook doesn't do anything for MachinePool, only the schema validation is shared.